tensorflow-framework/src/main/java/org/tensorflow/framework/callbacks/ProgbarLogger.java

```diff
@@ -0,0 +1,353 @@
+   *     this list will be logged as-is. All others will be averaged over time (e.g. loss, etc). If
+   *     not provided, defaults to the Model's metrics.
+   */
+  public ProgbarLogger(ProgressBar.CountMode mode, String... statefulMetrics) {
+    this(null, null, mode, Arrays.asList(statefulMetrics));
+  }
+
+  /**
+   * Create a ProgbarLogger
+   *
+   * @param params Training parameters
+   */
+  public ProgbarLogger(Map<String, Object> params) {
+    this(params, null, ProgressBar.CountMode.SAMPLES, (List<String>) null);
+  }
+
+  /**
+   * Create a ProgbarLogger
+   *
+   * @param params Training parameters
+   * @param model Reference of the model being trained.
+   */
+  public ProgbarLogger(Map<String, Object> params, Object model) {
+    this(params, model, ProgressBar.CountMode.SAMPLES, (List<String>) null);
+  }
+
+  /**
+   * Create a ProgbarLogger
+   *
+   * @param params Training parameters
+   * @param model Reference of the model being trained.
+   * @param mode Whether the progress bar should count SAMPLES seen or STEPS (batches) seen.
+   * @param statefulMetrics names of metrics that should not be averaged over an epoch. Metrics in
+   *     this list will be logged as-is. All others will be averaged over time (e.g. loss, etc). If
+   *     not provided, defaults to the Model's metrics.
+   */
+  public ProgbarLogger(
+      Map<String, Object> params,
+      Object model,
+      ProgressBar.CountMode mode,
+      String... statefulMetrics) {
+    this(params, model, mode, Arrays.asList(statefulMetrics));
+  }
+
+  /**
+   * Create a ProgbarLogger
+   *
+   * @param params Training parameters
+   * @param model Reference of the model being trained.
+   * @param unit Whether the progress bar should count SAMPLES seen or STEPS (batches) seen.
+   * @param statefulMetrics names of metrics that should not be averaged over an epoch. Metrics in
+   *     this list will be logged as-is. All others will be averaged over time (e.g. loss, etc). If
+   *     not provided, defaults to the Model's metrics.
+   */
+  public ProgbarLogger(
+      Map<String, Object> params,
+      Object model,
+      ProgressBar.CountMode unit,
+      List<String> statefulMetrics) {
+    // TODO super(params, model);
+
+    this.unit = unit;
+    this.statefulMetrics =
+        statefulMetrics != null ? new HashSet<>(statefulMetrics) : new HashSet<>();
+    setParams(params);
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public final void setParams(Map<String, Object> params) {
+    if (params == null) {
+      return;
+    }
+    super.setParams(params);
+    verbose =
+        ((ProgressBar.VerboseMode) params.getOrDefault("verbose", ProgressBar.VerboseMode.VERBOSE));
+    epochs = (Integer) params.getOrDefault("epochs", 1);
+    target =
+        unit == ProgressBar.CountMode.STEPS
+            ? (Integer) params.get("steps")
+            : (Integer) params.get("samples");
+    writer = (PrintWriter) params.get("writer");
```

Comment:
Should this throw if the writer isn't present?

Also the doc should note the other params and their names that it's expecting to find. Maybe make them public static final constants on this class?